.github/actions/deploy_v2/action.yml

```diff
@@ -0,0 +1,132 @@
+name: deploy_v2
+description: Deploys application to AKS
+
+inputs:
+  arm-access-key:
+    required: true
+  azure-credentials:
+    required: true
+  environment:
+    description: Environment to deploy to
+    required: true
+  pr-number:
+    description: PR number for the review app
+    required: false
+  sha:
+    description: Commit sha to be deployed
+    required: true
+  slack-webhook:
+    required: true
+
+outputs:
+  deploy-url:
+    value: ${{ steps.set_env_var.outputs.deploy_url }}
+
+runs:
+  using: composite
+  steps:
+    - name: Set Environment variables
+      id: set_env_var
+      shell: bash
+      run: |
+        tf_vars_file=terraform/workspace_variables/${{ inputs.environment }}.tfvars.json
+        cluster=$(jq -r '.cluster' ${tf_vars_file})
+
+        if [ -n "${{ inputs.pr-number }}" ]; then
+          echo "DEPLOY_ENV=review_aks-${{ inputs.pr-number }}" >> $GITHUB_ENV
+          echo "deploy_url=https://apply-review-${{ inputs.pr-number }}.${cluster}.teacherservices.cloud" >> $GITHUB_OUTPUT
+        else
+          echo "DEPLOY_ENV=${{ inputs.environment }}" >> $GITHUB_ENV
+
+          hostname=$(jq -r '.service_gov_uk_host_names[0]' ${tf_vars_file})
+          if [[ $hostname != null ]]; then
+            echo "deploy_url=https://${hostname}.apply-for-teacher-training.service.gov.uk" >> $GITHUB_OUTPUT
+          else
+            if [[ $cluster == 'production' ]]; then
+              echo "deploy_url=https://apply-${{ inputs.environment }}.teacherservices.cloud" >> $GITHUB_OUTPUT
+            else
+              echo "deploy_url=https://apply-${{ inputs.environment }}.${cluster}.teacherservices.cloud" >> $GITHUB_OUTPUT
+            fi
+          fi
+        fi;
+        echo "DOCKER_IMAGE=$DOCKER_IMAGE" >> $GITHUB_ENV
+
+        echo "KEY_VAULT_NAME=$(jq -r '.key_vault_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "KEY_VAULT_APP_SECRET_NAME=$(jq -r '.key_vault_app_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+        echo "KEY_VAULT_INFRA_SECRET_NAME=$(jq -r '.key_vault_infra_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+      env:
+        DOCKER_IMAGE: ${{ format('ghcr.io/dfe-digital/apply-teacher-training:{0}', inputs.sha) }}
+
+    - name: Use Terraform v1.2.3
+      uses: hashicorp/setup-terraform@v2
+      with:
+        terraform_version: 1.2.3
+
+    - uses: azure/login@v1
+      with:
+        creds: ${{ inputs.azure-credentials }}
+
+    - name: Validate Azure Key Vault secrets
+      uses: DFE-Digital/github-actions/validate-key-vault-secrets@master
+      with:
+        KEY_VAULT: ${{ env.KEY_VAULT_NAME }}
+        SECRETS: |
+          ${{ env.KEY_VAULT_APP_SECRET_NAME }}
+          ${{ env.KEY_VAULT_INFRA_SECRET_NAME }}
+
+    # - name: Delete clock to force its recreation
```

Comment:
We will need to implement this step in the future?  Have we got a ticket to test whether deleting the clock app is something we'll still need to do in AKS?

Reply:
Clock has not been deployed yet. It will be as part of this card: https://trello.com/c/lobIH6vR
This comment can be removed as hopefully we won't need to delete.